test/contrib/automotive/someip.uts

```diff
@@ -416,3 +272,451 @@ p.set_optionArray([SDOption_IP4_EndPoint(), SDOption_IP4_EndPoint()])
 
 assert(struct.unpack("!L", bytes(p)[4:8])[0] == 32)
 assert(struct.unpack("!L", bytes(p)[40:44])[0] == 24)
+
+
++ Git issue 2348: SOME/IP-SD Entry-Array is broken by building it from RAW
+
+= Single SD entry
+# offer service
+ea1 = SDEntry_Service()
+ea1.type = 1
+ea1.srv_id = 0x1234
+ea1.inst_id = 0x5678
+ea1.ttl = 0x333333
+
+# find service
+ea2 = SDEntry_Service()
+ea2.type = 0
+ea2.srv_id = 0x8765
+ea2.inst_id = 0x4321
+ea2.ttl = 0x222222
+
+sd1 = SD()
+sd1.set_entryArray([ea1])
+# this is computed on build, but we need it sooner for the assert()
+sd1.len_entry_array = 16
+sd1.len_option_array = 0
+
+assert(sd1.show(dump=True) == SD(sd1.build()).show(dump=True))
```

Comment:
What is this ? What are you trying to test ?

Reply:
This verifies that parsing works correctly, which was the problem in #2348. I construct a packet, then dump it and parse it again, the result from show() should be identical

Reply:
Alright i see what you did there. We usually never test the result of `show()` because it isn't static, but i guess it will do in this case